File: e2e-apps/java-spring-web/scripts/fixture_driver.py
#!/usr/bin/env python3

import json
import sys
from pathlib import Path
from urllib.parse import urlsplit


def main() -> int:
    if len(sys.argv) < 3:
        print("usage: fixture_driver.py <prepare|auth-state> <fixture> [output_path] [entry_url]", file=sys.stderr)
        return 1

    action = sys.argv[1]
    fixture = sys.argv[2]

    if action == "prepare":
        print(f"prepared fixture {fixture}")
        return 0

    if action == "auth-state":
        if len(sys.argv) < 4:
            print("auth-state requires an output path", file=sys.stderr)
            return 1
        output_path = Path(sys.argv[3])
        entry_url = sys.argv[4] if len(sys.argv) > 4 else "http://127.0.0.1:3201"
        parsed_url = urlsplit(entry_url)
        origin = f"{parsed_url.scheme}://{parsed_url.netloc}" if parsed_url.scheme and parsed_url.netloc else "http://127.0.0.1:3201"
        output_path.parent.mkdir(parents=True, exist_ok=True)
        state = {
            "cookies": [
                {
                    "name": "healer_session",
                    "value": fixture,
                    "domain": "127.0.0.1",
                    "path": "/",
                    "httpOnly": False,
                    "secure": False,
                    "sameSite": "Lax",
                }
            ],
            "origins": [
                {
                    "origin": origin,
                    "localStorage": [
                        {"name": "fixture_profile", "value": fixture},
                    ],
                }
            ],
        }
        output_path.write_text(json.dumps(state, indent=2), encoding="utf-8")
        print(f"wrote auth state for {fixture}")
        return 0

    print(f"unsupported action: {action}", file=sys.stderr)
    return 1


if __name__ == "__main__":
    raise SystemExit(main())

File: e2e-apps/ruby-rails-web/scripts/fixture_driver.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require "json"
require "fileutils"
require "uri"

action = ARGV[0].to_s
fixture = ARGV[1].to_s

if action.empty? || fixture.empty?
  warn "usage: fixture_driver.rb <prepare|auth-state> <fixture> [output_path] [entry_url]"
  exit 1
end

case action
when "prepare"
  puts "prepared fixture #{fixture}"
when "auth-state"
  output_path = ARGV[2].to_s
  entry_url = ARGV[3].to_s
  if output_path.empty?
    warn "auth-state requires an output path"
    exit 1
  end

  FileUtils.mkdir_p(File.dirname(output_path))
  origin =
    begin
      uri = URI.parse(entry_url.empty? ? "http://127.0.0.1:3101" : entry_url)
      "#{uri.scheme}://#{uri.host}:#{uri.port}"
    rescue URI::InvalidURIError
      "http://127.0.0.1:3101"
    end
  state = {
    cookies: [
      {
        name: "healer_session",
        value: fixture,
        domain: "127.0.0.1",
        path: "/",
        httpOnly: false,
        secure: false,
        sameSite: "Lax"
      }
    ],
    origins: [
      {
        origin: origin,
        localStorage: [
          { name: "fixture_profile", value: fixture }
        ]
      }
    ]
  }
  File.write(output_path, JSON.pretty_generate(state))
  puts "wrote auth state for #{fixture}"
else
  warn "unsupported action: #{action}"
  exit 1
end
